sockeye/inference.py

```diff
@@ -306,11 +314,11 @@ def run_encoder(self,
                                 provide_data=self._get_encoder_data_shapes(source_max_length))
 
         self.encoder_module.forward(data_batch=batch, is_train=False)
-        decoder_states = self.encoder_module.get_outputs()
+        decoder_init_states = self.encoder_module.get_outputs()
 
         # replicate encoder/init module results beam size times
-        decoder_states = [mx.nd.repeat(s, repeats=self.beam_size, axis=0) for s in decoder_states]
-        return ModelState(decoder_states)
+        decoder_init_states = [mx.nd.repeat(s, repeats=self.beam_size, axis=0) for s in decoder_init_states]
```

Comment:
I think this is where we should 'extract' the predicted lengths/length ratios and hide this complexity entirely from the Translator. I think it would make sense to expand ModelState to have another member (length/length_ratio) that is optional.

Reply:
Added estimated_reference_Length to ModelState

Reply:
Moved now it to Inference Model.